uint32 distanceFromAnchor
int32 AnchorNodeId
uint32 qf